### PR TITLE
feat(selling): add delivery window to delivery stops

### PR DESCRIPTION
**Pull request by prafful1234**
Depends on https://github.com/DigiThinkIT/erpnext/pull/328.

added default delivery window to delivery settings

default settings are fetched if delivery window not set for customer.

### CODE REVIEW

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- erpnext/stock/doctype/delivery_trip/delivery_trip.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DigiThinkIT/erpnext/pullRequest?prid=5110492&bid=16804764)

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py [hunk 1]
```diff
@@ -251,6 +251,15 @@ def get_directions(self, route, optimize):
...
 
+@frappe.whitelist()
+def get_delivery_window(customer=None):
+	delivery_window = frappe.db.get_value("Customer", customer, ["delivery_start_time", "delivery_end_time"], as_dict=1)
+	delivery_window.default_window = False
+	if delivery_window and (delivery_window.delivery_start_time or delivery_window.delivery_end_time):
+		return delivery_window
+	default_window = frappe.db.get_value("Delivery Settings","Delivery Settings",["delivery_start_time", "delivery_end_time"],as_dict=1)
+	default_window.default_window = True
+	return default_window
```
Comment (Alchez) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.py]:
```suggestion
	delivery_start_time = delivery_start_time = None
	fields = ["delivery_start_time", "delivery_end_time"]

	if customer:
		delivery_start_time, delivery_start_time = frappe.db.get_value("Customer", customer, fields)
	else:
		delivery_start_time, delivery_end_time = frappe.db.get_value("Delivery Settings", "Delivery Settings", fields)

	return {
		"delivery_start_time": delivery_start_time,
		"delivery_end_time": delivery_start_time,
		"default_window": bool(customer)
	}
```

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- erpnext/stock/doctype/delivery_trip/delivery_trip.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DigiThinkIT/erpnext/pullRequest?prid=5110492&bid=16804764)

**Review by Alchez — DISMISSED** [review]

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- erpnext/stock/doctype/delivery_trip/delivery_trip.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DigiThinkIT/erpnext/pullRequest?prid=5110492&bid=16804764)

**Review by Alchez — APPROVED** [review]

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- erpnext/stock/doctype/delivery_trip/delivery_trip.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DigiThinkIT/erpnext/pullRequest?prid=5110492&bid=16804764)

**Pull request merged**